### ISSUE TITLE
🐛 the `--files-ignore-unknown` flag doesn't ignore files without extensions when used with `--stdin-file-path`

### ISSUE THREAD

**Issue by adamhl8** [biomejs/biome#6566]
### Environment information

```block
CLI:
  Version:                      2.0.5
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.6.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.2.17

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

```sh
echo "contents" | biome check --stdin-file-path=/path/to/file --files-ignore-unknown=true --write
```

Running this gives the following error:

```
✖ The file passed via --stdin-file-path doesn't have an extension. Biome needs a file extension to know how handle the file.
```

### Expected result

Biome should ignore files without extensions when  `--stdin-file-path` is used with `--files-ignore-unknown=true`.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
The `--files-ignore-unknown=true` works for files only (no directories). Which means that the flag should only work when `--stdin-file-path` is a path to an extension that Biome doesn't know how to handle.

In your case, --stdin-file-path might be a folder, and that's why there's an error. This is expected

**Comment by adamhl8** [comment]
In my specific case where I ran into this error, the path in question was one to a real file. But that doesn't matter, right? Because `--stdin-file-path` only needs the extension, the file doesn't even need to exist.

I think it would make sense for `--stdin-file-path` to always assume the given path is a file. I can't imagine a situation where it would be a folder because then what content would you be piping in?

**Comment by ematipico** [comment]
> In my specific case where I ran into this error, the path in question was one to a real file. But that doesn't matter, right? Because `--stdin-file-path` only needs the extension, the file doesn't even need to exist.
> 
> I think it would make sense for `--stdin-file-path` to always assume the given path is a file. I can't imagine a situation where it would be a folder because then what content would you be piping in?

Yeah, I suppose we could be less strict and just return the original content if the file doesn't have an extension. The reason I added the error is that there were users who passed files without extensions, and expected that to work.
